content/blog/2022/2022-08-03-feature-flags-in-nodejs-react-client-with-launchdarkly.md

```diff
@@ -0,0 +1,320 @@
+
+For a client-side or mobile SDK to evaluate our feature flags, the SDK needs to authenticate against the LaunchDarkly server. According to the [documentation](https://docs.launchdarkly.com/home/getting-started/feature-flags#making-flags-available-to-client-side-and-mobile-sdks), we need to enable the *"SDKs using Client-side ID"* option for when we want to access feature flags from a React UI or any Javascript UI. We will therefore retrieve the *Client-Side ID* from the LaunchDarkly "Projects" page and paste it into our code. Additionally, we need to confirm that the flag's targeting option is turned on after creating the flag.
+
+## Creating a Simple React App
+
+Let’s start by creating a simple React UI application. We will build this using one of the most popular npm libraries, [`create-react-app`](https://create-react-app.dev/). It is one of the simplest methods to start a new React project, making it a great option for both serious, large-scale apps and your own, individual projects.
+
+We will use the tool `npx` which gives us the ability to use the `create-react-app` package without having to first install it on our computer, which is very convenient. It also ensures that we are always using the latest version of `create-react-app`:
+
+```bash
+npx create-react-app nodejs-react-feature-flag-launchdarkly
+```
+
+This will create a folder with the name `nodejs-react-feature-flag-launchdarkly` and automatically install all the required packages inside that folder. The folder tree would look something like this:
+
+```bash
+nodejs-react-feature-flag-launchdarkly
+├── README.md
+├── node_modules
+├── package.json
+├── .gitignore
+├── public
+└── src
+```
+
+Now we can simply run our app with this command:
+
+```bash
+npm start
+```
+
+The development server will start up at http://localhost:3000 and we can see the home page:
+
+{{% image alt="React UI" src="images/posts/nodejs-frontend-launchdarkly/React_Logo.png" %}}
+
+Next, we add the [`launchdarkly-js-client-sdk`](https://github.com/launchdarkly/js-client-sdk) package to use client-side methods to fetch the feature flag variations:
+
+```bash
+npm install launchdarkly-js-client-sdk
+```
+
+## Toggling a Feature on the Fly
+
+In our first use case, let’s try to simply define a Boolean feature flag in the LaunchDarkly UI. 
+
+We will simply define a feature button in the UI and show it to the user or not, depending on the targeting rules defined in Launchdarkly.
+
+The feature flag we define in LaunchDarkly UI looks something like this:
+
+{{% image alt="String Feature Flag LaunchDarkly" src="images/posts/nodejs-frontend-launchdarkly/Hide_User.png" %}}
```

Comment:
@arpendu11 this image still shows the old feature flag before your changes. Can you update it?

Reply:
Updated the screenshot.